libvast/src/segment_store.cpp

```diff
@@ -352,12 +345,34 @@ void segment_store::inspect_status(caf::settings& dict) {
   put(current, "size", builder_.table_slice_bytes());
 }
 
-segment_store::segment_store(path dir, uint64_t max_segment_size,
-                             size_t in_memory_segments)
-  : dir_{std::move(dir)},
-    max_segment_size_{max_segment_size},
-    cache_{in_memory_segments} {
-  // nop
+caf::error segment_store::register_segments() {
+  for (auto filename : directory{segment_path()})
+    if (auto err = register_segment(filename))
```

Comment:
We should probably _not_ break out of the loop here. An unrelated file that happens to be in this folder should not break the application.

Reply:
If we actually get an unrelated file, things break really bad when we call this a few lines before after mmaping the file:

```cpp
  auto s = fbs::GetSegment(chk->data());
```

I've added an extra check that bails out when the flatbuffer integrity is compromised.